tests/python/pants_test/util/test_objects.py

```diff
@@ -199,15 +199,29 @@ def test_validate(self):
     collection_exactly_a_or_b = TypedCollection(Exactly(self.A, self.B))
     self.assertEqual([self.A()], collection_exactly_a_or_b.validate_satisfied_by([self.A()]))
     self.assertEqual([self.B()], collection_exactly_a_or_b.validate_satisfied_by([self.B()]))
-    with self.assertRaisesWithMessage(
-        TypeConstraintError,
-        "in wrapped constraint TypedCollection(Exactly(A or B)): value A() (with type 'A') must satisfy this type constraint: SubclassesOf(Iterable)."):
+    with self.assertRaisesWithMessage(TypeConstraintError, """\
+in wrapped constraint TypedCollection(Exactly(A or B)): value A() (with type 'A') must satisfy this type constraint: SubclassesOf(Iterable).
+Note that objects matching SubclassesOf(str) are not considered iterable."""):
       collection_exactly_a_or_b.validate_satisfied_by(self.A())
-    with self.assertRaisesWithMessage(
-        TypeConstraintError,
-        "in wrapped constraint TypedCollection(Exactly(A or B)) matching iterable object [C()]: value C() (with type 'C') must satisfy this type constraint: Exactly(A or B)."):
+    with self.assertRaisesWithMessage(TypeConstraintError, """\
+in wrapped constraint TypedCollection(Exactly(A or B)) matching iterable object [C()]: value C() (with type 'C') must satisfy this type constraint: Exactly(A or B)."""):
       collection_exactly_a_or_b.validate_satisfied_by([self.C()])
 
+  def test_iterable_detection(self):
+    class StringCollectionField(datatype([('hello_strings', TypedCollection(Exactly(text_type)))])):
+      pass
+
+    self.assertEqual(['xxx'], StringCollectionField(hello_strings=['xxx']).hello_strings)
+
+    with self.assertRaisesWithMessage(TypeCheckError, """\
+type check error in class StringCollectionField: errors type checking constructor arguments:
```

Comment:
Pluralising `errors` here suggests I'm looking for more than one in the following text, but I can only see one?

Reply:
I will convert this to `error(s)`!